File: Modules/SEDO/EMS/SIE.Web.EMS/Equipments/Boms/Commands/InsertChildSparePartCommand.cs
using SIE.Web.Command;

namespace SIE.Web.EMS.Equipments.Boms.Commands
{
    /// <summary>
    /// 插入子数据命令
    /// </summary>
    public class InsertChildSparePartCommand : ViewCommand
    {
       /// <summary>
       /// 执行
       /// </summary>
       /// <param name="args"></param>
       /// <param name="scope"></param>
       /// <returns></returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            return true;
        }
    }
}

File: Modules/SEDO/EMS/SIE.Web.EMS/Equipments/Boms/Commands/InsertChildSparePartCommand.js
SIE.defineCommand('SIE.Web.EMS.Equipments.Boms.Commands.InsertChildSparePartCommand', {
    extend: 'SIE.cmd.Add',
    meta: { text: "添加子", group: "edit", iconCls: "icon-FileTree icon-green" },
    canExecute: function (view) {
        var entity = view.getCurrent();
        if (entity == null)
            return false;
        return true;
    },
    execute: function (view, source) {
        var childNode = view.insertNewChild();
        childNode.generateId();
        childNode.setEquipBomId(view.getParent().getCurrent().data.Id);
        view.getControl().setSelection(childNode); 
    }
});

